fe/lib/components/LocationSuggest/LocationSuggest.tsx

```diff
@@ -132,25 +132,29 @@ export const LocationSuggest = forwardRef<HTMLInputElement, Props>(
     }, [suggestData]);
 
     useEffect(() => {
-      if (nearestLocationsData) {
+      if (nearestLocationsData && nearestLocationsData.nearestLocations) {
         // The closest SEEK location returned for the geolocation input
         const closestLocation = nearestLocationsData.nearestLocations[0];
 
-        const {
-          contextualName,
-          id: { value },
-        } = closestLocation;
+        const { contextualName } = closestLocation;
 
         setPlaceholder(contextualName);
-        setSelectedLocationId(value);
+
+        if (onSelect) {
+          onSelect(closestLocation);
+        }
+
+        if (closestLocation?.id?.value) {
+          setSelectedLocationId(closestLocation.id.value);
+        }
```

Comment:
```suggestion
        handleLocationSelect(closestLocation)
```

You can remove this duplicate code and just use `handleLocationSelect`

Reply:
Updated ✅